yodeploy/hooks/configurator.py

```diff
@@ -97,10 +97,23 @@ def write_config_js(self):
         """
         if not self.public_config_js_path:
             return
+
         path = self.deploy_path(self.public_config_js_path)
         pub_conf_json = json.dumps(self.pub_config)
-        with open(path, 'r') as f:
+        token = self.public_config_token
+
+        with open(path, 'rb') as f:
             content = f.read()
-        content = content.replace(self.public_config_token, pub_conf_json)
+
+        # In python 3, we need to coerce the content string to be the same
+        # type as token and pub_conf_json before we can do our token
+        # substitution. The following converts content to a unicode string.
```

Comment:
We totally can (which I didn't realize) and I've gone back and experimented with that approach. If we go that route, that implementation ends up looking like this instead:

```python
        pub_conf_json = json.dumps(self.pub_config)  # unicode py3, bytes py2
        token = self.public_config_token  # unicode py3, bytes py2

        try:  # Python 2: Convert pub_conf_json and token to unicode
            pub_conf_json = pub_conf_json.decode('utf-8')
            token = token.decode('utf-8')
        except AttributeError:
            pass

        with codecs.open(path, 'r', 'utf-8') as f:
            content = f.read()  # unicode in python 2 and 3

        # everything is unicode at this point in all envs
        content = content.replace(token, pub_conf_json)
```

As it was written originally the try/except makes all three strings unicode under python 3 and keeps all three strings byte strings under python 2. If we change it to this, `content` starts off as unicode in both 2 and 3 (:+1: for consistency), but then we have the opposite problem and we need to change `token` and `pub_conf_json` to unicode in python 2. After looking at this again today, I'm sure I'm missing something obvious I'm just not sure what.